### PR TITLE
GT-1830 Upgrade Okta authentication to use Combine Publishers and UseCases instead of shared Observables

### PR DESCRIPTION
**Pull request by levieggertcru**
This PR updates OktaAuthentication to use Combine Publishers and UseCases for authenticating. 

This PR includes the following changes:

- Moved ```EmailSignUpService.swift``` to Share/Data/EmailSignUpService and instantiation into AppDataLayerDependencies.swift.
- Removed OktaAuthentication.swift dependency from EmailSignUpService.swift.  Instead the AuthenticateUserUseCase will attempt to post email sign ups when a user authenticates. 
- Removed OktaAuthentication.swift from FirebaseAnalytics.swift and SnowplowAnalytics.swift.  Instead the AuthenticateUserUseCase will set user properties when a user authenticates.
- Added CruOktaAuthentication+GetAuthenticationInstance extension for fetching a new CruOktaAuthentication instance from AppDataLayerDependencies.
- Added CruOktaAuthentication+GetAuthUser extension for fetching the CruOktaUserDataModel.
- Added UseCases AuthenticateUserUseCase, LogOutUserUseCase, GetUserIsAuthenticatedUseCase, GetUserAccountProfileNameUseCase.
- In AppFlow attempt to renew okta authorization on app launches via AuthenticateUserUseCase.
- Removed OktaAuthentication.swift and associated Observables.  Using CruOktaAuthentication.swift directly provided by OktaAuthentication swift module.
- Added GetUserAccountProfileNameUseCase to AccountViewModel for fetching user profile name.  Previously AccountViewModel was accessing OktaAuthentication.

### CODE REVIEW

File: godtools/App/Share/Data/CruOktaAuthentication/CruOktaAuthentication+GetAuthenticationInstance.swift [hunk 1]
```diff
@@ -1,32 +1,32 @@
...
-    func configureAndCreateNewOktaAuthentication(appBuild: AppBuild) -> CruOktaAuthentication {
-                
         let signInPath: String = "auth"
         let signOutPath: String = "auth/logout"
         let redirectBaseUrl: String
         
+        // TODO: Change back redirectBaseUrl. ~Levi
+        /*
         if appBuild.isDebug {
             redirectBaseUrl = "org.cru.godtools.debug"
```
Comment (levieggertcru) [inline godtools/App/Share/Data/CruOktaAuthentication/CruOktaAuthentication+GetAuthenticationInstance.swift:23]: @rachaelblue I'm going to create a ticket for this TODO.  For some reason ```org.cru.godtools.debug``` stopped working.

File: godtools/App/Features/Menu/Presentation/Menu/MenuViewModel.swift [hunk 2]
```diff
@@ -74,24 +72,21 @@ class MenuViewModel: NSObject, MenuViewModelType {
...
-        }
+            .store(in: &cancellables)
     }
     
     private func reloadMenuDataSource() {
         
-        let isAuthorized: Bool = oktaUserAuthentication.isAuthenticated
+        let isAuthorized: Bool = getUserIsAuthenticatedUseCase.getUserIsAuthenticated()
         
         // TODO: Disabling the account section in the menu by hardcoding to false.
```
Comment (levieggertcru) [inline godtools/App/Features/Menu/Presentation/Menu/MenuViewModel.swift:91]: Hey @rachaelblue also note here in ```MenuViewModel``` that the sign in section is hard coded to be disabled.  You'll need to set that in order to test authentication.

**Review by rachaelblue — COMMENTED** [review]
✅ was able to log in and out after enabling the menu option

File: godtools/App/Features/Menu/Domain/UseCases/AuthenticateUserUseCase/AuthenticateUserAuthType.swift [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+//  AuthenticateUserAuthType.swift
+//  godtools
+//
+//  Created by Levi Eggert on 10/24/22.
+//  Copyright © 2022 Cru. All rights reserved.
+//
+
+import UIKit
+
+enum AuthenticateUserAuthType {
```
[the remark this answers is not in the document]
Comment (levieggertcru) [reply godtools/App/Features/Menu/Domain/UseCases/AuthenticateUserUseCase/AuthenticateUserAuthType.swift:11]: Yes that should be.

File: godtools/App/Features/Menu/Domain/UseCases/AuthenticateUserUseCase/AuthenticateUserUseCase.swift [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+                        .mapError { oktaError in
+                            return oktaError.getError()
+                        }
+                        .eraseToAnyPublisher()
+                })
+                .eraseToAnyPublisher()
+        }
+    }
+    
+    func authenticatePublisher(authType: AuthenticateUserAuthType) -> AnyPublisher<Bool, Error> {
```
[the remark this answers is not in the document]
Comment (levieggertcru) [reply godtools/App/Features/Menu/Domain/UseCases/AuthenticateUserUseCase/AuthenticateUserUseCase.swift:62]: That is a bit confusing, I'm going to make a name change to the private method.

**Review by rachaelblue — APPROVED** [review]

**Pull request merged**